src/main/resources/assets/admin/common/js/form/inputtype/text/TextInputType.ts

```diff
@@ -50,37 +51,43 @@ export abstract class TextInputType
         (<FormInputEl>occurrence).resetBaseValues();
     }
 
-    protected initOccurrenceListeners(inputEl: FormInputEl) {
+    protected initOccurrenceListeners(inputEl: FormInputEl): void {
         if (this.hasMaxLengthSet() || this.showTotalCounter) {
-            const counterEl: InputValueLengthCounterEl = new InputValueLengthCounterEl(inputEl, this.maxLength, this.showTotalCounter);
+            new InputValueLengthCounterEl(inputEl, this.maxLength, this.showTotalCounter);
         }
 
-        return inputEl;
+        inputEl.onValueChanged((event: ValueChangedEvent) => {
+            this.handleOccurrenceInputValueChanged(inputEl);
+        });
     }
 
     createInputOccurrenceElement(index: number, property: Property): Element {
         if (!this.getValueType().equals(property.getType())) {
             ValueTypeConverter.convertPropertyValueType(property, this.getValueType());
         }
 
-        const inputEl: FormInputEl = this.createInput(index, property);
-
-        inputEl.onValueChanged((event: ValueChangedEvent) => {
-            this.handleOccurrenceInputValueChanged(inputEl);
-        });
-
-        inputEl.setSpellcheck(true);
-        const lang = this.getContext().formContext.getLanguage();
-        if (!StringHelper.isEmpty(lang)) {
-            const langs = lang.split('-');
-            inputEl.setLang(langs.length > 1 ? langs[0] : lang);
-        }
+        const inputEl: FormInputEl = <FormInputEl>this.createInput(index, property).setSpellcheck(true);
 
+        this.updateInputLangParams(inputEl);
         this.initOccurrenceListeners(inputEl);
 
         return inputEl;
     }
 
+    protected updateInputLangParams(inputEl: FormInputEl): void {
+        const locale: string = this.getContext().formContext.getLanguage();
+
+        if (!StringHelper.isEmpty(locale)) {
+            const language: string = Locale.extractLanguage(locale);
+            inputEl.setLang(language);
+
+            if (Locale.isRtlCode(language)) {
+                inputEl.getEl().setAttribute('dir', 'rtl'); // doesn't work since reset.less in enonic-admin-artifacts...
```

Comment:
Let's implement `setDir`, similar to `setLang`. `.style.direction` can be set there as well.